elastic/kibana
# Adding FIPS configuration option and validation

## Summary


Kibana administrators need to way to configure their Kibana so that the application knows its running in 'FIPS Mode' 

By setting `xpack.security.fipsMode.enabled` to `true`, Kibana will know it's intended to run in FIPS mode. This configuration will be used in the future to enable/disable functionality and to verify correct licensing.

Also, in order to run in FIPS mode, the Node.js environment must be configured to use a FIPS 140-2 compliant OpenSSL3 provider and have FIPS enabled.

This config is available for OnPrem and ECK.

## Testing Instructions

Add `xpack.security.fipsMode.enabled: true` to your `kibana.yml` and start Kibana, you will see an error logged and the Kibana process will stop with error code 78.

Testing the alternate failure path (`xpack.security.fipsMode.enabled: false`, Node FIPS enabled) is not available at this time as there are still existing uses of non-compliant algorithms which will cause Kibana to fail on startup.

## Questions

Ops team: Im struggling to get the os-builds to work so I can `docker build`, I keep getting this error, any tips?

│ERROR failure 0 sec
   │ERROR Error: Premature close
   │          at Socket.onclose (node:internal/streams/end-of-stream:154:30)
   │          at Socket.emit (node:events:526:35)
   │          at Pipe.<anonymous> (node:net:337:12)

I'd like to verify that this new config option is proper translated from a Docker Env Var
